tests/interlocked_hl_int/interlocked_hl_int.c

```diff
@@ -0,0 +1,287 @@
+// Copyright(C) Microsoft Corporation.All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+
+#include <inttypes.h>
+#include <stdbool.h>
+#include <stdlib.h>
+
+
+#include "testrunnerswitcher.h"
+
+#include "c_pal/threadapi.h"
+#include "macro_utils/macro_utils.h"
+
+#include "c_util/interlocked_hl.h"
+
+TEST_DEFINE_ENUM_TYPE(THREADAPI_RESULT, THREADAPI_RESULT_VALUES);
+TEST_DEFINE_ENUM_TYPE(INTERLOCKED_HL_RESULT, INTERLOCKED_HL_RESULT_VALUES);
+
+BEGIN_TEST_SUITE(TEST_SUITE_NAME_FROM_CMAKE)
+
+TEST_SUITE_INITIALIZE(suite_init)
+{
+}
+
+TEST_SUITE_CLEANUP(suite_cleanup)
+{
+}
+
+TEST_FUNCTION_INITIALIZE(function_initialize)
+{
+}
+
+TEST_FUNCTION_CLEANUP(function_cleanup)
+{
+}
+
+volatile_atomic int32_t globalValue = 10;
+
+/*
+Tests:
+InterlockedHL_WaitForValue
+InterlockedHL_DecrementAndWake
+*/
+static int decrement_and_wake_helper_thread_function(void* context)
+{
+    (void)context;
+    INTERLOCKED_HL_RESULT result = InterlockedHL_WaitForValue(&globalValue, 9, UINT32_MAX);
+    ASSERT_ARE_EQUAL(INTERLOCKED_HL_RESULT, INTERLOCKED_HL_OK, result);
+    return 0;
+}
+
+TEST_FUNCTION(interlocked_hl_decrement_and_wake_operates_successfully)
+{
+    // + have a helper thread which waits for decremented value
+    // + main thread creates helper thread and decrements value
+    // + helper thread wakes up when the global address value is decremented by 1
+    // arrange
+    globalValue = 10;
+    THREAD_HANDLE helper_thread_handle;
+    THREADAPI_RESULT thread_result = ThreadAPI_Create(&helper_thread_handle, decrement_and_wake_helper_thread_function, NULL);
+    ASSERT_ARE_EQUAL(THREADAPI_RESULT, THREADAPI_OK, thread_result);

```

Comment:
Can have the assert in the same line with the call, so you can remove the thread_result variable alltogether